src/Symfony/Component/HttpClient/HttpOptions.php

```diff
@@ -0,0 +1,299 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient;
+
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * A helper providing autocompletion for available options.
+ *
+ * @see HttpClientInterface for a description of each options.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @experimental in 4.3
+ */
+class HttpOptions
+{
+    private $options = [];
```

Comment:
We could add a constructor `__construct(array $options = [])`. That would allow us to use the class to modify an existing set of options.

Reply:
Clone to the rescue? Wouldn't it be a better fit?

Reply:
Depends. Would I use `HttpOptions` to store options for later use or would I store options as an array? I the former case, `clone` would help but I would be missing getters on this class. In the latter case, `clone` is useless.

Reply:
Not sure to get the use case but let's consider after merge, sure!